# Aquisição, Armazenamento e Recuperação de Dados em Larga Escala

## Basic Information

Título do Projeto: Sistema de Recolha e Recuperação de Notícias

Alunos:
- Semahattin Can Doruk (nº 57321)
- Efşah Memişoğlu (nº 57322)
- João Maravilhoso (nº 50996)

## Background and Motivation

O objetivo deste projeto é recolher e permitir pesquisar notícias online através de palavras-chave.

In [15]:
import sys
sys.path.append(".")

from information_retrieval import search_articles, print_results

## Consulta 1 - Visualização Geral

In [16]:
results = search_articles("all", limit=5)
print_results(results)


Total de artigos carregados: 244
Resultados apresentados: 5

1. No title
Engine: bbc_turkce
Date: 2026-04-27T11:34:27
URL: https://www.bbc.com/turkce/articles/cy41919qxn9o

2. No title
Engine: bbc_turkce
Date: 2026-04-27T11:34:27
URL: https://www.bbc.com/turkce/articles/cg73735j54mo

3. No title
Engine: bbc_turkce
Date: 2026-04-27T11:34:27
URL: https://www.bbc.com/turkce/articles/cr41437672qo

4. No title
Engine: bbc_turkce
Date: 2026-04-27T11:34:27
URL: https://www.bbc.com/turkce/articles/cly9yd9k3q9o

5. No title
Engine: bbc_turkce
Date: 2026-04-27T11:34:27
URL: https://www.bbc.com/turkce/articles/cgrlr10448wo


Esta consulta mostra alguns artigos recolhidos para validar os dados armazenados.

## Consulta 2 - Pesquisa por palavra-chave

In [17]:
results = search_articles("haber", limit=5)
print_results(results)


Total de artigos carregados: 244
Resultados apresentados: 5

1. No title
Engine: bundle_gundem
Date: 2026-04-27T11:34:23
URL: https://haberglobal.com/gundem/gulistan-doku-sorusturmasi-donemin-valisi-tuncay-sonelin-konusmasi-ortaya-cikti-520980?utm_campaign=Bundle&utm_medium=referral&utm_source=Bundle

2. No title
Engine: bundle_gundem
Date: 2026-04-27T11:34:23
URL: https://www.dunya.com/gundem/iran-ile-yurutulen-ateskes-gorusmeleri-cikmaza-girdi-trump-gorusmeyi-iptal-etti-haberi-822904?utm_campaign=Bundle&utm_medium=referral&utm_source=Bundle

3. No title
Engine: bundle_gundem
Date: 2026-04-27T11:34:23
URL: https://www.dunya.com/ekonomi/google-guney-korede-yapay-zeka-kampusu-kuruyor-haberi-823053?utm_campaign=Bundle&utm_medium=referral&utm_source=Bundle

4. No title
Engine: bundle_gundem
Date: 2026-04-27T11:34:23
URL: https://www.dunya.com/foto-galeri/ekonomi/baharin-habercisi-tezgaha-indi-fiyati-cep-yaksa-da-uretici-memnun-galeri-823027?utm_campaign=Bundle&utm_medium=referral&utm_sou

Esta consulta procura artigos que contenham a palavra "haber".

## Consulta 3 - Pesquisa por palavra-chave

In [18]:
results = search_articles("turk", limit=5)
print_results(results)


Total de artigos carregados: 244
Resultados apresentados: 5

1. No title
Engine: bbc_turkce
Date: 2026-04-27T11:34:27
URL: https://www.bbc.com/turkce/articles/cy41919qxn9o

2. No title
Engine: bbc_turkce
Date: 2026-04-27T11:34:27
URL: https://www.bbc.com/turkce/articles/cg73735j54mo

3. No title
Engine: bbc_turkce
Date: 2026-04-27T11:34:27
URL: https://www.bbc.com/turkce/articles/cr41437672qo

4. No title
Engine: bbc_turkce
Date: 2026-04-27T11:34:27
URL: https://www.bbc.com/turkce/articles/cly9yd9k3q9o

5. No title
Engine: bbc_turkce
Date: 2026-04-27T11:34:27
URL: https://www.bbc.com/turkce/articles/cgrlr10448wo


Esta consulta demonstra a recuperação de informação relacionada com o termo "turk".

## Resultados

Foram carregados 244 artigos a partir dos ficheiros JSON gerados pelos scrapers. Os resultados apresentados demonstram que os dados foram corretamente armazenados e podem ser recuperados de forma estruturada, incluindo informação como fonte, data de recolha e URL.

## Avaliação dos Resultados

O sistema demonstrou ser capaz de recuperar informação relevante a partir dos dados recolhidos. As consultas realizadas permitiram obter artigos relacionados com os termos pesquisados, demonstrando a funcionalidade do sistema.

No entanto, verificou-se que alguns artigos não apresentam título, o que indica limitações no processo de extração de dados. Apesar disso, o sistema apresenta um bom desempenho geral.

## Melhorias Futuras

Como melhorias futuras, seria interessante garantir a extração completa de todos os campos dos artigos, especialmente o título.

Além disso, poderiam ser implementadas técnicas de ranking para ordenar os resultados por relevância, bem como a integração com ferramentas como Elasticsearch para permitir pesquisas mais avançadas e eficientes.

## Declaração de Integridade Académica

Nos , declaramos  ter desenvolvido o presente trabalho em conformidade com o Código de Integridade da Universidade da Beira Interior, não tendo incorrido em qualquer forma de fraude académica.

## Declaração de Uso de IA

Ferramentas utilizadas:
- ChatGPT

Utilização:
As ferramentas de inteligência artificial foram utilizadas como apoio à programação, esclarecimento de dúvidas e melhoria da organização do código. Todo o conteúdo foi compreendido e validado pelos autores.

## Conclusão

O presente projeto permitiu desenvolver um sistema funcional de aquisição, armazenamento e recuperação de dados a partir de fontes de notícias online.

Através de técnicas de web scraping, foi possível recolher um conjunto significativo de artigos, posteriormente armazenados em formato JSON. A implementação de mecanismos de pesquisa por palavras-chave demonstrou ser eficaz na recuperação de informação relevante.

Apesar de algumas limitações, como a ausência de certos campos em alguns artigos, o sistema apresenta uma base sólida e extensível. Como trabalho futuro, poderão ser integradas técnicas mais avançadas de recuperação de informação para melhorar a qualidade dos resultados.